src/adapter/dummyadapter.c

```diff
@@ -77,15 +77,15 @@ static bool DummyAdapter_CreateIdTransaction(DIDAdapter *_adapter, const char *p
     }
 
     DIDDocument *doc = DIDRequest_FromJson(&info->request, root);
-    if (!doc)
+    if (!doc && strcmp(info->request.header.op, "deactivate"))
```

Comment:
Please check this condition.
In my guess, it would be "(!doc || strcmp(info->request.header.op, "deactivate") != 0)".